### PR TITLE
 hreflang alternate

### LINKED ISSUE #1086: Missing language in meta hreflang if language is not default language
Example:

Default language: DE
Other languages: EN, IT

DE:

```html
<link rel="alternate" href="http://www.example.com/de" hreflang="de" />
<link rel="alternate" href="http://www.example.com/de" hreflang="x-default" />
<link rel="alternate" href="http://www.example.com/en" hreflang="en" />
<link rel="alternate" href="http://www.example.com/it" hreflang="it" />
```

EN:

```html
<link rel="alternate" href="http://www.example.com/de" hreflang="de" />
<link rel="alternate" href="http://www.example.com/de" hreflang="x-default" />
<link rel="alternate" href="http://www.example.com/it" hreflang="it" />
```

In EN the language EN is missing.

### PR DESCRIPTION
**Pull request by alexander-schranz**
Show correct alternative links and remove invalid meta tag

__Tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes
- [ ] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | none
| Fixed tickets | fixes #1086
| BC Breaks     | no
| Doc           | none

### CODE REVIEW

File: src/Sulu/Bundle/WebsiteBundle/Twig/Meta/MetaTwigExtension.php [hunk 1]
```diff
@@ -129,7 +126,6 @@ public function getSeoMetaTags($extension, $content)
 
         // build meta tags
         $result = array();
-        $result[] = $this->getMeta('title', $seo['title']);
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/WebsiteBundle/Twig/Meta/MetaTwigExtension.php]: not a valid w3c meta tag.
Searchengines use <title></title> Tag

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@ CHANGELOG for Sulu
 ==================
 
 * dev-develop
+    * BUGFIX      #1172 [ContentBundle]  Fix hreflang meta tags, remove unvalid title meta tag
```
Comment (danrot) [inline CHANGELOG.md]: @alexander-schranz It's invalid, not unvalid ;)

**Comment by danrot** [comment]
It would be nice to have some tests here, so that this will be right anytime in the future...

**Comment by chirimoya** [comment]
@alexander-schranz ... can you please provide some tests. We should merge this PR. THX.

**Comment by alexander-schranz** [comment]
@danrot @chirimoya tests for the meta twig extension added

File: src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+use PHPCR\SessionInterface;
+use Sulu\Bundle\WebsiteBundle\Twig\Content\ContentPathInterface;
+use Sulu\Bundle\WebsiteBundle\Twig\Meta\MetaTwigExtension;
+use Sulu\Component\Content\Structure;
+use Sulu\Component\Localization\Localization;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Sulu\Component\Webspace\Portal;
+use Sulu\Component\Webspace\Webspace;
+
+
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php]: no new line

File: src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    /**
+     * @var ContentPathInterface
+     */
+    private $contentPath;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->requestAnalyzer = $this->prophesize('Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface');
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php]: Use `RequestAnalyzer::class` instead. Makes later restructuring easier.

File: src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        $locale->setLanguage('en');
+
+        $portal = new Portal();
+        $portal->setDefaultLocalization($locale);
+
+        $this->requestAnalyzer->getWebspace()->willReturn($webspace);
+        $this->requestAnalyzer->getPortal()->willReturn($portal);
+        $this->requestAnalyzer->getCurrentLocalization()->willReturn($locale);
+
+        $this->contentPath = $this->prophesize('Sulu\Bundle\WebsiteBundle\Twig\Content\ContentPathInterface');
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php]: Also use `ContentPathInterface::class` here.

File: src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        ));
+
+        $this->assertEquals(
+            array(
+                '<link rel="alternate" href="/de/test" hreflang="de" />',
+                '<link rel="alternate" href="/en/test-en" hreflang="x-default" />',
+                '<link rel="alternate" href="/en/test-en" hreflang="en" />',
+                '<link rel="alternate" href="/en/test-en-us" hreflang="en-us" />',
+                '<link rel="alternate" href="/fr/test-fr" hreflang="fr" />',
+            ), explode(PHP_EOL, $urls)
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php]: explode into an own line

File: src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+                'title' => 'Page Title'
+            )
+        );
+
+        $this->assertEquals(
+            array(
+                '<meta name="description" content="Excerpt Description">',
+                '<meta name="keywords" content="SEO, Keywords">',
+                '<meta name="robots" content="INDEX, FOLLOW">',
+            ), explode(PHP_EOL, $metaTags)
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Twig/MetaTwigExtensionTest.php]: explode into own line

**Pull request merged**